### ISSUE TITLE
Chat disappears unless an editor has the document opened

### ISSUE THREAD

**Issue by ar27111994** [ONLYOFFICE/DocumentServer#553]
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
If two users open a document simultaneously with one user having edit permission edit and the other user having view permission, then chat gets reset if editor leaves the document, even though the other user is still online with the document opened.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Two users open a document simultaneously with one user having edit permission edit and the other user having view permission
**What is the expected behavior?**
chat should not get reset unless all users involved in it regardless of their permissions close the document
**Did this work in previous versions of DocumentServer?**
don't know
**DocumentServer version:**
5.2.8
**Operating System:**
Cent OS
**Browser version:**
71.0.3578.80

**Comment by XDmitryK** [comment]
Hi
Reset chat history occurs after each assembly of the document.
Keeping the chat history is not possible due to the design of the application.

**Comment by ar27111994** [comment]
But why doesn't chat disappear/reset if an editor has a document opened and another user leaves the document and re-opens after some time? It should disappear in that case too, or should not disappear if only viewers have opened document. Meaning behavior should be consistent, right?

**Comment by XDmitryK** [comment]
Chat history visible for all user when is joined document in current session. but if all users has left of document that chat history will be reset. There is no difference, the user in viewing mode or the user in editing mode
Also, if the user reconnects to the document in the current session, he will see the entire chat history.

**Comment by ar27111994** [comment]
Agreed. But this isn't happening when the editor leaves document and comes back again the chat is reset for editor, even though the viewer had the document opened all this time. Please look into and see if you can reproduce the issue as described above.

**Comment by XDmitryK** [comment]
I was wrong 
Anyone who is in view mode needs to update the document after assembling the document in order to get the current chat history.

**Comment by ar27111994** [comment]
ok, what does that mean? :-) Please elaborate.

**Comment by XDmitryK** [comment]
Well, it is conceived in the application. The actions that I have indicated above are considered correct.)
User in view mode is not the active user in the document
The assembly of the document may begin when all users exit editing mode, users in viewing mode are not considered. They will have to update the document every time.

**Comment by ar27111994** [comment]
So is there any workaround regarding this bug?

**Comment by XDmitryK** [comment]
I passed your message to the design department. They will think about it.
It remains to wait for their decision.
